libsolidity/parsing/Token.cpp

```diff
@@ -101,14 +109,14 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-unsigned Token::extractM(string const& _literal)
+unsigned Token::extractUnsigned(string::const_iterator _begin, string::const_iterator _end)
 {
 	try
 	{
-		unsigned short m = stoi(_literal);
+		unsigned short m = boost::lexical_cast<unsigned short>(boost::make_iterator_range(_begin, _end));
 		return m;
 	}
-	catch(out_of_range& e)
+	catch(const boost::bad_lexical_cast &)
 	{
 		return 0;
```

Comment:
I disagree with the name change...this also handles the sizes for uint and bytes...should probably change it to something like ```parseSize``` . I'll throw in some tests for the identifier and types for good measure though.